components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -126,6 +127,50 @@ clean {
     delete GLEAN_BOOTSTRAP_DIR
 }
 
+// Support for extracting metrics.yaml from artifact files.
+
+// This is how to extract `metrics.yaml` and `pings.yaml` from AAR files: an "artifact transform"
+// identifies the files in an "exploded AAR" directory.  See
+// https://docs.gradle.org/current/userguide/dependency_management_attribute_based_matching.html#sec:abm_artifact_transforms.
+// This is exactly how elements of AAR files are consumed by the Android-Gradle plugin; see the
+// transforms defined in
+// https://android.googlesource.com/platform/tools/base/+/studio-master-dev/build-system/gradle-core/src/main/java/com/android/build/gradle/internal/dependency/AarTransform.java
+// and their usage at
+// https://android.googlesource.com/platform/tools/base/+/studio-master-dev/build-system/gradle-core/src/main/java/com/android/build/gradle/internal/VariantManager.java#592.
+//
+// Note that this mechanism only applies to `module` dependencies (i.e., AAR files downloaded from
+// Maven) and not to `project` dependencies in the same root project or substituted as part of a
+// Gradle composite build.
+class GleanMetricsYamlTransform extends ArtifactTransform {
+    List<File> transform(File file) {
+        def f = new File(file, "metrics.yaml")
+        if (f.exists()) {
+            println("Glean SDK - extracting ${f.path} from an AAR archive")
```

Comment:
You'll get nicer output if you get a `logger` from a Gradle `Project` instance -- `logger.lifecycle` is often what you want for "always on" output, or `logger.info` or `logger.debug`.  There are examples of this in patches I posted, if you care to do it.

Reply:
Oh, thanks, good catch.